pkg/opensearch/response_parser_test.go

```diff
@@ -0,0 +1,1030 @@
+					"alias": "{{term @host}} {{metric}} and {{not_exist}} {{@host}}",
+					"metrics": [{ "type": "count", "id": "1" }],
+		 "bucketAggs": [
+						{ "type": "terms", "field": "@host", "id": "2" },
+						{ "type": "date_histogram", "field": "@timestamp", "id": "3" }
+					]
+				}`,
+		}
+		response := `{
+		   "responses": [
+			 {
+			   "aggregations": {
+				 "2": {
+				   "buckets": [
+					 {
+					   "3": {
+						 "buckets": [{ "doc_count": 1, "key": 1000 }, { "doc_count": 3, "key": 2000 }]
+					   },
+					   "doc_count": 4,
+					   "key": "server1"
+					 },
+					 {
+					   "3": {
+						 "buckets": [{ "doc_count": 2, "key": 1000 }, { "doc_count": 8, "key": 2000 }]
+					   },
+					   "doc_count": 10,
+					   "key": "server2"
+					 },
+					 {
+					   "3": {
+						 "buckets": [{ "doc_count": 2, "key": 1000 }, { "doc_count": 8, "key": 2000 }]
+					   },
+					   "doc_count": 10,
+					   "key": 0
+					 }
+				   ]
+				 }
+			   }
+			 }
+		   ]
+		}`
+		rp, err := newResponseParserForTest(targets, response)
+		assert.Nil(t, err)
+		result, err := rp.getTimeSeries()
+		assert.Nil(t, err)
+		assert.Len(t, result.Responses, 1)
+
+		queryRes := result.Responses["A"]
+		assert.NotNil(t, queryRes)
+		assert.Len(t, queryRes.Frames, 3)
+
+		seriesOne := queryRes.Frames[0]
+		assert.Equal(t, "server1 Count and {{not_exist}} server1", seriesOne.Name)
+		assert.Len(t, seriesOne.Fields, 2)
+		assert.Equal(t, time.Date(1970, time.January, 1, 0, 0, 1, 0, time.UTC), *seriesOne.Fields[0].At(0).(*time.Time))
+		assert.Equal(t, time.Date(1970, time.January, 1, 0, 0, 2, 0, time.UTC), *seriesOne.Fields[0].At(1).(*time.Time))
+		assert.EqualValues(t, 1, *seriesOne.Fields[1].At(0).(*float64))
+		assert.EqualValues(t, 3, *seriesOne.Fields[1].At(1).(*float64))
+
+		seriesTwo := queryRes.Frames[1]
+		assert.Equal(t, "server2 Count and {{not_exist}} server2", seriesTwo.Name)
+		assert.Len(t, seriesTwo.Fields, 2)
+		assert.Equal(t, time.Date(1970, time.January, 1, 0, 0, 1, 0, time.UTC), *seriesTwo.Fields[0].At(0).(*time.Time))
+		assert.Equal(t, time.Date(1970, time.January, 1, 0, 0, 2, 0, time.UTC), *seriesTwo.Fields[0].At(1).(*time.Time))
+		assert.EqualValues(t, 2, *seriesTwo.Fields[1].At(0).(*float64))
+		assert.EqualValues(t, 8, *seriesTwo.Fields[1].At(1).(*float64))
+
+		seriesThree := queryRes.Frames[2]
+		assert.Equal(t, "0 Count and {{not_exist}} 0", seriesThree.Name)
+		assert.Len(t, seriesThree.Fields, 2)
+		assert.Equal(t, time.Date(1970, time.January, 1, 0, 0, 1, 0, time.UTC), *seriesThree.Fields[0].At(0).(*time.Time))
+		assert.Equal(t, time.Date(1970, time.January, 1, 0, 0, 2, 0, time.UTC), *seriesThree.Fields[0].At(1).(*time.Time))
+		assert.EqualValues(t, 2, *seriesThree.Fields[1].At(0).(*float64))
+		assert.EqualValues(t, 8, *seriesThree.Fields[1].At(1).(*float64))
+	})
+
+	// TODO: this test will require some conversion of tables to data frames, original work in Elasticsearch https://github.com/grafana/grafana/pull/34710
```

Comment:
nit: It may make sense to also have the number of the opensearch issue here

Reply:
done